src/main/java/com/exasol/containers/ContainerSynchronizationVerifier.java

```diff
@@ -0,0 +1,55 @@
+package com.exasol.containers;
+
+import static com.exasol.containers.ExasolContainerConstants.MAX_ALLOWED_CLOCK_OFFSET_IN_MILLIS;
+
+import org.testcontainers.containers.ContainerLaunchException;
+
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * Check the synchronization of a given container in relation to the host running the Test Container software.
+ */
+public class ContainerSynchronizationVerifier {
+    private final ContainerTimeService timeService;
+
+    /**
+     * Create a new {@link ContainerSynchronizationVerifier}.
+     *
+     * @param timeService service that provide the time in the container
+     * @return new {@link ContainerSynchronizationVerifier} instance
+     */
+    public static ContainerSynchronizationVerifier create(final ContainerTimeService timeService) {
+        return new ContainerSynchronizationVerifier(timeService);
+    }
+
+    /**
+     * Create a synchronization verifier for a given container.
+     *
+     * @param timeService service that provides the container time
+     */
+    public ContainerSynchronizationVerifier(final ContainerTimeService timeService) {
```

Comment:
```suggestion
    ContainerSynchronizationVerifier(final ContainerTimeService timeService) {
```
Should only be used in unit test

Reply:
Made private. There is not much point in having it open when you have a factory method.